core/src/main/java/org/apache/metamodel/query/AggregateFunction.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.metamodel.query;
+
+import org.apache.metamodel.util.AggregateBuilder;
+
+/**
+ * Interface that contains the aggregation specific methods
+ * related to the AggregateBuilder.
+ *
+ */
+public interface AggregateFunction extends FunctionType {
+
+    public AggregateBuilder<?> createAggregateBuilder();
+
+    public Object evaluate(Object... values);
```

Comment:
Not blocking for me right now (since we can easily add it), but I would prefer some javadoc on those methods. Especially to also point out that the evaluate method is just a shorthand for creating an aggregate builder, adding all the values and then calculating the value. Maybe that even warrants removing the method, but maybe you don't want to do that as part of this refactoring.